agent/agent.go

```diff
@@ -1061,13 +1068,13 @@ func setEndTaskFailureDetails(tc *taskContext, detail *apimodels.TaskEndDetail,
 
 }
 
-func (a *Agent) killProcs(ctx context.Context, tc *taskContext, ignoreTaskGroupCheck bool, reason string) {
+func (a *Agent) killProcs(ctx context.Context, tc *taskContext, ignoreTaskGroupCheck bool, reason string, additionalShouldKillCheck bool) {
```

Comment:
Rather than add an additional flag that we have to set to false everywhere, did we consider just checking the `taskConfig` is set or not in `shouldKill`? It seems like just checking `taskConfig` for nilness in `shouldKill` is logically the same (i.e. it'll return early) without needing to introduce a lot of plumbing of one special case value.

Reply:
If I do that, I won't be able to use nt and the initial taskConfig:

> taskGroupIsNil := nt.TaskGroup == ""
		if initialTC != nil && initialTC.taskConfig != nil {
			taskGroupIsNil = initialTC.taskConfig.TaskGroup == nil
		}
		
Which is what was happening before we [simplified task config](https://github.com/evergreen-ci/evergreen/pull/6985/files#diff-62b6ad581fe3a3059ae8c85ef0f31dde4092bfdecfa7d6857c470bcacaa8cc8b)

Reply:
Hmm, it’s true that if the task config is nil, shouldKill will return true anyway. So this wouldn’t help the case were we shouldn’t be killing procs (because it’s part of a task group), but we lost that information by the tc not being populated. I'll remove this.